File: Models/ProductViewModel.cs
namespace ProductCatalogAppSii.Models
{
    public class ProductViewModel
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public decimal Price { get; set; }
        public string Description { get; set; }
        public Category Category { get; set; } // Reutilizamos la clase Category de Product
        public string Image { get; set; }
        public decimal Tax { get; set; }
        public decimal PriceWithTax { get; set; }
    }
}

File: ProductCatalogApp.AngularApp/src/app/models/product-view-model.ts

export interface Category {
  id: number;
  name: string;
  slug: string;
  image: string;
  creationAt: string; 
  updatedAt: string;  
}

export interface ProductViewModel {
  id: number;
  title: string;       
  slug: string;
  price: number;
  description: string;
  category: Category;  
  images: string[];    
  creationAt: string;  
  updatedAt: string;  
}
